plugin/core/sessions.py

```diff
@@ -220,24 +275,49 @@ def clear_dotted_value(current: dict, dotted: str) -> None:
     current.pop(keys[-1], None)
 
 
-class Session(object):
-    def __init__(self,
-                 config: ClientConfig,
-                 workspace_folders: List[WorkspaceFolder],
-                 client: Client,
-                 on_pre_initialize: 'Optional[Callable[[Session], None]]' = None,
-                 on_post_initialize: 'Optional[Callable[[Session], None]]' = None,
-                 on_post_exit: Optional[Callable[[str], None]] = None) -> None:
+class Session(Client):
+
+    @classmethod
+    def name(cls) -> str:
+        raise NotImplementedError()
+
+    @classmethod
+    def needs_update_or_installation(cls) -> bool:
```

Comment:
This is not called by the code. As is `install_or_update`.
This will make it impossible to implement NpmClientHandler for new API right?

Reply:
Well, `NpmClientHandler` can of course keep doing things how it's doing them currently and handle that itself. So is the plan to implement that later?

Reply:
> So is the plan to implement that later?

Yes.